[Nixon] iP

nixonwidjaja opened the pull request:
# AdaBot

> "Adapt. Improvise. Overcome." - Tom Cruise

AdaBot frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,
1. Add your tasks
2. Let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [x] Managing tasks
- [x] Managing deadline
- [ ] Reminders (coming soon)

##
If you Java programmer, you can use it to practice Java too. Here's the ```main``` method in ```Duke.java```:
```java
 public static void main(String[] args) {
     new Duke().run();
 }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

nixonwidjaja committed:
Rename, add greet and exit

nixonwidjaja committed:
Add echo

nixonwidjaja committed:
Add tasks and list command

nixonwidjaja committed:
Add Mark as Done

nixonwidjaja committed:
Add todo, events, deadlines

nixonwidjaja committed:
Add DukeException and handle errors

nixonwidjaja committed:
Delete task

nixonwidjaja committed:
Try Automated Text UI Testing

nixonwidjaja committed:
Add file handler to save task list

nixonwidjaja committed:
Merge pull request #1 from nixonwidjaja/branch-Level-7

Level-7

nixonwidjaja committed:
Enforce local date to deadline

nixonwidjaja committed:
Merge pull request #2 from nixonwidjaja/branch-Level-8

Enforce local date to deadline

nixonwidjaja committed:
Enforce LocalDate on event

nixonwidjaja committed:
Fix bug preserve mark on file

nixonwidjaja committed:
Stretch goal: Print events on specific date

nixonwidjaja committed:
Extract out Ui, Storage, Parser, TaskList

nixonwidjaja committed:
Move queryList to Parser

nixonwidjaja committed:
Organize into packages

nixonwidjaja committed:
Merge remote-tracking branch 'origin/add-gradle-support' into add-packages

nixonwidjaja committed:
Setup Gradle

nixonwidjaja committed:
Merge pull request #3 from nixonwidjaja/add-packages

Organize into packages and setup Gradle

nixonwidjaja committed:
Add find functionality

nixonwidjaja committed:
Fix directories

nixonwidjaja committed:
Merge pull request #4 from nixonwidjaja/branch-Level-9

Fix directories

nixonwidjaja committed:
Add JavaDoc

nixonwidjaja committed:
Merge pull request #5 from nixonwidjaja/branch-A-JavaDoc

Add JavaDoc

nixonwidjaja committed:
Add JavaDoc for classes

nixonwidjaja committed:
Merge pull request #6 from nixonwidjaja/branch-A-CodingStandard

Add JavaDoc for classes

nixonwidjaja committed:
Add JUnit tests

nixonwidjaja committed:
Update README.md

nixonwidjaja committed:
Update README.md

nixonwidjaja committed:
Fix checkstyle

nixonwidjaja committed:
Merge branch 'master' of https://github.com/nixonwidjaja/ip

nixonwidjaja committed:
Refactor logic to Command classes

teozern1 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,22 @@
+/**
+ * This command functions to add task to a task list.
+ */
+public class AddCommand extends Command {
```
I like how all the commands have the same naming convention

teozern1 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,274 @@
+public class Parser {
+    private static String emptyDeadlineDate = "Please specify deadline date using /by\n"
+        + "e.g. deadline report /by 2023-12-31";
+    private static String emptyEventDate = "Please specify event start and end dates using /from and /to\n"
```
Seems good to give an example maybe I should do that too

teozern1 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,124 @@
+     * @param date Date given by user
+     * @throws DukeException if invalid date
+     */
+    public void date(TaskList tasks, String date) throws DukeException {
```
Maybe could be more specific in naming, mentioning that the date is already converted? Might potentially cause confusion in the future

teozern1 reviewed (commented):
I like the usage of the parser using switch and cases to parse the commands! Looks very clean. The classes are also organized into different packages, will definitely try implementing that in my own ip.

nixonwidjaja committed:
First iteration for GUI

nixonwidjaja committed:
Fix checkstyle

nixonwidjaja committed:
Integrate Duke with GUI

nixonwidjaja committed:
Merge pull request #7 from nixonwidjaja/branch-Level-10

Integrate GUI

samuelim01 commented on .gitignore:
```diff
@@ -15,3 +15,5 @@ bin/
 /text-ui-test/ACTUAL.TXT
 text-ui-test/EXPECTED-UNIX.TXT
+/data
+/src/main/java/duke/data
```
Good job excluding the other files!

samuelim01 commented on build.gradle:
```diff
@@ -0,0 +1,63 @@
+}
+
+application {
+    mainClass.set("duke.Duke")
```
This should be changed to use the new GUI!
```suggestion
    mainClass.set("duke.Launcher")
```

samuelim01 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,135 @@
+    /**
+     * Displays bye message.
+     */
+    public String bye() {
```
I think you could benefit from using clearer method names in the `Ui` class
```suggestion
public String getByeMessage() {
```

samuelim01 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,274 @@
+            case "todo":
+                return new ToDo(q[1], q[2].equals("X"));
+            default:
+                throw new DukeException("Error parsing file data");
```
Perhaps you could create more exceptions that inherit from `DukeException` such as `FileParseException`!

samuelim01 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,274 @@
+        List<String> queryList = convertToList(input);
+        switch (queryList.get(0)) {
+
+        case "bye":
```
Nice, you followed the code standard for switch-case!

samuelim01 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,62 @@
+ * Represents a deadline inherits from a Task.
+ */
+public class Deadline extends Task {
+    private static String invalidDate = "Please provide date with the following format: YYYY-MM-DD";
```
This could be made final
```suggestion
    private static final String INVALID_DATE = "Please provide date with the following format: YYYY-MM-DD";
```

samuelim01 commented on src/main/java/duke/ui/Ui.java:
I think most of the methods here could be made static?

samuelim01 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,274 @@
+     * @return Event instance
+     * @throws DukeException if no event description, invalid format, or invalid date
+     */
+    private static Event parseUserEvent(List<String> queryList) throws DukeException {
```
This method is slightly long and unclear. Perhaps you could add a comment or two to explain what you are doing here. 😃

samuelim01 commented on src/main/java/gui/MainWindow.java:
```diff
@@ -0,0 +1,76 @@
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    private String getResponse(String input) {
```
It's a bit strange to put this method here, perhaps you could move it into `Duke` instead?

samuelim01 reviewed (commented):
LGTM! Just a few nits, but otherwise nice bot!

nixonwidjaja committed:
Assert query list must have at least 1 element
Query list with 0 length is an invalid state.

Asserting query list length > 1 help improve code maintainability and app correctness

nixonwidjaja committed:
Change list name for consistency

nixonwidjaja committed:
Switch assert place

nixonwidjaja committed:
Merge pull request #8 from nixonwidjaja/A-Assertions

A-Assertions

nixonwidjaja committed:
Fix bugs
Resize chatbox to accommodate many text

Add command to list all command

Couple of other bug fixes

nixonwidjaja committed:
Fix checkstyle

nixonwidjaja committed:
Merge pull request #9 from nixonwidjaja/A-CodeQuality

Fix bugs

nixonwidjaja committed:
C-FriendlierSyntax

nixonwidjaja committed:
Fix from code review

nixonwidjaja committed:
Fix bugs on already marked tasks

nixonwidjaja committed:
Fix overparameter bug

nixonwidjaja committed:
Add user guide link

nixonwidjaja committed:
Add UI screenshot

nixonwidjaja committed:
Add detailed user guide

nixonwidjaja committed:
Add user guide config

nixonwidjaja committed:
Merge branch 'master' of https://github.com/nixonwidjaja/ip

nixonwidjaja committed:
Fix table

nixonwidjaja committed:
Remove config

nixonwidjaja committed:
Fix jar cannot run

nixonwidjaja committed:
Fix checkstyle

nixonwidjaja committed:
Update README.md

nixonwidjaja committed:
Update README.md

nixonwidjaja committed:
Update README.md

nixonwidjaja committed:
Acknowledge code reuse

nixonwidjaja committed:
Merge branch 'master' of https://github.com/nixonwidjaja/ip

nixonwidjaja committed:
Update jdk version
